neptune/new/internal/threading/daemon.py

```diff
@@ -53,3 +61,35 @@ def run(self):
     @abc.abstractmethod
     def work(self):
         pass
+
+    class ConnectionRetryWrapper:
+        INITIAL_RETRY_BACKOFF = 2
+        MAX_RETRY_BACKOFF = 20
+
+        def __call__(self, func):
+            @functools.wraps(func)
+            def wrapper(self_: Daemon, *args, **kwargs):
+                # pylint: disable=protected-access
+                while not self_._interrupted:
+                    try:
+                        result = func(self_, *args, **kwargs)
+                        if self_.last_backoff_time > 0:
+                            self_.last_backoff_time = 0
+                            click.echo("Communication with Neptune restored!", sys.stderr)
+                        return result
+                    except NeptuneConnectionLostException:
+                        if self_.last_backoff_time == 0:
+                            click.echo(f"Experiencing connection interruptions. "
+                                       f"Will try to reestablish communication with Neptune.",
+                                       sys.stderr)
+                            self_.last_backoff_time = self.INITIAL_RETRY_BACKOFF
+                        else:
+                            self_.last_backoff_time = min(self_.last_backoff_time * 2, self.MAX_RETRY_BACKOFF)
+                        time.sleep(self_.last_backoff_time)
+                    except Exception:
+                        click.echo("Unexpected error occurred. Killing Neptune asynchronous thread. "
+                                   "All data is safe on disk.",
```

Comment:
This warning message is related to async processor only and should not be here in Daemon.

Reply:
I wanted to reuse it for ping (because before it just died, loudly and with a stacktrace). If I tweak the message between async proc and ping it would be ok?